From: Liang Yan <lyan@suse.com>
Date: Thu, 30 May 2019 17:20:06 -0400
Subject: KVM: s390: kABI Workaround for 'lowcore'
Patch-mainline: Never, kABI workaround
References: bsc#1119222

Add avg_steal_timer and remove pad_ox318 in lowcore. 
Hide that from the kABI checker.
Related patch:
s390-vtime-steal-time-exponential-moving-average.patch

Signed-off-by: Liang Yan <lyan@suse.com>

---
 arch/s390/include/asm/lowcore.h |    4 ++++
 1 file changed, 4 insertions(+)

--- a/arch/s390/include/asm/lowcore.h
+++ b/arch/s390/include/asm/lowcore.h
@@ -98,7 +98,11 @@ struct lowcore {
 
 	/* Current process. */
 	__u64	current_task;			/* 0x0328 */
+#ifndef __GENKSYMS__
 	__u64	avg_steal_timer;		/* 0x0330 */
+#else
+	__u8    pad_0x318[0x320-0x318];         /* 0x0330 */
+#endif	
 	__u64	kernel_stack;			/* 0x0338 */
 
 	/* Interrupt, DAT-off and restartstack. */
